lib/fifo.ts

```diff
@@ -1,12 +1,19 @@
 import StaticFIFO from "./static.ts";
 
+export const END = Symbol("Stream ended.");
+export const ERROR = Symbol("Stream errored.");
+
+export type Enqueueable<T> = T | Promise<T> | typeof END | typeof ERROR;
```

Comment:
Allowing `Promise<T>` to be queued might lead a user to think the queue will spit out the resolved value, which it won't. I think this should also just be left up to usage, i.e. not including `Promise<T>` here, and rather let the user do something like `const fifo = new FIFO<number | Promise<number>>` themselves if they want to.